--- /datas2/moodles/moodle3.0.10/moodle/repository/filepicker.js	2017-05-07 21:32:44.000000000 +0200
+++ /datas2/gitworkspace/moodle27/moodle2_version/repository/filepicker.js	2017-06-15 09:33:28.673857147 +0200
@@ -1032,10 +1032,13 @@
                 return;
             }
             this.active_repo.nextpagerequested = true;
-            var nextpage = this.active_repo.page+1;
+            var nextpage = this.active_repo.page+1; 
             var args = {
                 page: nextpage,
-                repo_id: this.active_repo.id
+                repo_id: this.active_repo.id,
+                //Unistra Dev
+                s: ((this.active_repo.searchtext!= undefined && this.active_repo.searchtext !=null && this.active_repo.searchtext!='') ?this.active_repo.searchtext:'')
+                //End Unistra Dev
             };
             var action = this.active_repo.issearchresult ? 'search' : 'list';
             this.request({
@@ -1438,6 +1441,9 @@
             this.active_repo.message = (data.message || '');
             this.active_repo.help = data.help?data.help:null;
             this.active_repo.manage = data.manage?data.manage:null;
+            //UNISTRA Dev
+            this.active_repo.searchtext = data.searchtext?data.searchtext:null;
+            //End UNISTRA Dev
             this.print_header();
         },
         print_login: function(data) {
@@ -1841,6 +1847,9 @@
                     scope: this,
                     action:'searchform',
                     repository_id: this.active_repo.id,
+                    //Unistra dev
+                    search_text: this.active_repo.searchtext,
+                    //end Unistra dev
                     callback: function(id, obj, args) {
                         if (obj.repo_id == scope.active_repo.id && obj.form) {
                             // if we did not jump to another repository meanwhile
@@ -1848,6 +1857,11 @@
                             // Highlight search text when user click for search.
                             var searchnode = searchform.one('input[name="s"]');
                             if (searchnode) {
+                            	//Unistra dev
+                                if(this.search_text){
+                                	searchnode.set('value',this.search_text);
+                                }
+                                //End Unistra dev
                                 searchnode.once('click', function(e) {
                                     e.preventDefault();
                                     this.select();
